Sources/OutlineView/OutlineViewDragAndDrop.swift

```diff
@@ -0,0 +1,139 @@
+import Foundation
+import AppKit
+
+/// A protocol for use with `OutlineView`, implemented by an object (most likely
+/// the data source) to handle receiving items that are dragged into the `OutlineView`.
+@available(macOS 10.15, *)
+public protocol DropReceiver {
+    associatedtype DataElement: Identifiable
+    
+    /// Given a dragged `NSPasteboardItem`, attempts to read the data in the item
+    /// into a usable object for the `OutlineView`'s data source.
+    ///
+    /// - Parameter item: The Pasteboard item that is being dragged into the OutlineView
+    ///
+    /// - Returns: nil if the Pasteboard item is not readable by this receiver, or
+    ///   a tuple with the decoded item and its associated PasteboardType, which must
+    ///   be included in `acceptedTypes`
+    func readPasteboard(item: NSPasteboardItem) -> DraggedItem<DataElement>?
+    
+    /// Called continuously as a dragged item is moved over the `OutlineView`, this is
+    /// used to define the behavior of the `OutlineView` and the drag cursor.
+    ///
+    /// - Parameter target: A `DropTarget` object describing where the dragged item
+    ///   is being dragged over.
+    ///
+    /// - Returns: A case of `ValidationResult`, which will either highlight the area
+    ///   of the `OutlineView` where the item will be dropped, or some other behavior.
+    ///   See `ValidationResult` for possible return values.
+    func validateDrop(target: DropTarget<DataElement>) -> ValidationResult<DataElement>
+    
+    /// Called once the mouse is released during a drag into the `OutlineView`, after
+    /// `validateDrop(target:)` has returned a case other than `deny`, this function should
+    /// handle updating the data source
```

Comment:
Same here:

> Handles updating the data source once an item is dropped on the `OutlineView`. Called once after the drop completes and `validateDrop(target:)` returns a case other than `deny`.

Reply:
Got it here https://github.com/Sameesunkaria/OutlineView/pull/10/commits/9a66eb8d6576eb16ca2d046bf17d013ac32249e3